fsc/hdf5_io/_special_types.py

```diff
@@ -185,4 +185,4 @@ def _to_hashable(obj):
 
 @_to_hashable.register(Iterable)
 def _(obj):
-    return tuple(obj)
+    return tuple([_ensure_hashable(val) for val in obj])
```

Comment:
What is this change, how is it related to the optional attribute change?

Reply:
It's not directly related, but for the key to be hashable contents of e.g. a tuple need to be hashable also.

Reply:
ok